### PR TITLE
tests: divide all tests into the CI chunks including submodule tests

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #6181.

After #6183, the time cost for `chunks (10)` occasionally exceeds 10 minutes. This PR ensures that submodule tests can also be segmented into different chunks to enhance overall test speed.

### What is changed and how does it work?

```commit-message
Divide all tests into the CI chunks including submodule tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -216,12 +216,7 @@ basic-test: install-tools
 ci-test-job: install-tools dashboard-ui
 	@$(FAILPOINT_ENABLE)
 	CGO_ENABLED=1 go test -timeout=15m -tags deadlock -race -covermode=atomic -coverprofile=covprofile -coverpkg=./... $(shell ./scripts/ci-subtask.sh $(JOB_COUNT) $(JOB_INDEX))
-	@$(FAILPOINT_DISABLE)
-
-ci-test-job-submod: install-tools dashboard-ui
-	@$(FAILPOINT_ENABLE)
-	@ for mod in $(SUBMODULES); do cd $$mod && $(MAKE) ci-test-job && cd $(ROOT_PATH) > /dev/null && cat $$mod/covprofile >> covprofile; done
-	@$(FAILPOINT_DISABLE)
+	@ for mod in $(shell ./scripts/ci-subtask.sh $(JOB_COUNT) $(JOB_INDEX) 1); do cd $$mod && $(MAKE) ci-test-job && cd $(ROOT_PATH) > /dev/null && cat $$mod/covprofile >> covprofile; done
```
Comment (HuSharp) [inline Makefile:219]: Do we also need `@$(FAILPOINT_DISABLE)` or just by `clean`?

Comment (JmPotato) [reply Makefile:219]: This task is only used for the CI enviroment, so there is no need to do the clean-up work.

File: tests/integrations/client/Makefile [hunk 2]
```diff
@@ -12,37 +12,35 @@
...
+test: failpoint-enable
+	CGO_ENABLED=1 go test -v -tags deadlock -race -cover || { $(MAKE) failpoint-disable && exit 1; }
+	$(MAKE) failpoint-disable
 
-basic-test:
-	# skip
-
-ci-test-job: enable-codegen
-	CGO_ENABLED=1 go test -tags deadlock -race -covermode=atomic -coverprofile=covprofile -coverpkg=../../... github.com/tikv/pd/tests/client
+ci-test-job:
```
Comment (rleungx) [inline tests/integrations/client/Makefile:34]: Do we need to enable the failpoint?

Comment (JmPotato) [reply tests/integrations/client/Makefile:34]: It will be enabled by the root Makefile, so we don't need to enable it here again.

**Comment by rleungx** [comment]
I find that chunk 1 is much longer than before.

**Comment by JmPotato** [comment]
> I find that chunk 1 is much longer than before.

I manually hardcode the chunk index here, it seems that the speed is ok now.

**Review by nolouch — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**